sql/core.go

```diff
@@ -395,10 +395,19 @@ type FilteredTable interface {
 // that's more optimized given the columns that are projected.
 type ProjectedTable interface {
 	Table
-	WithProjections(colNames []string) Table
+	WithProjections([]string) Table
 	Projections() []string
 }
 
+type RelId uint16
+
+// RelationalNode is a namespaced source of rows
+type RelationalNode interface {
```

Comment:
You can do this on a second pass, but this interface definitely doesn't belong in the sql package and I would prefer that it didn't exist at all. It's only used during analysis, and then only during projection pushdown. Better to track this info via additional data structures local to that calculation, rather than making the node itself responsible for storage of this info.